database/src/main/java/com/firebase/ui/database/FirebaseIndexListAdapter.java

```diff
@@ -54,6 +56,6 @@ public FirebaseIndexListAdapter(Activity activity,
                                     @LayoutRes int modelLayout,
                                     Query keyRef,
                                     Query dataRef) {
-        super(activity, modelClass, modelLayout, new FirebaseIndexArray(keyRef, dataRef));
+        super(activity, modelClass, modelLayout, new FirebaseIndexArray(keyRef, dataRef.getRef()));
```

Comment:
@samtstern This shouldn't be a behavior change because we ignore ordering anyway. I wanted to make that clearer so `FirebaseIndexArray` only accepts a `DatabaseReference`, hence the need to convert the `Query`.